malaba6/E-Rental
# #167399298 built user registration endpoint

#### What does this PR do?
It creates a user registration endpoint.

#### Description of Task to be done?
1. It sets up the server
2. It creates authentication route file and specifies the user registration route.
3. It specifies the registration route in the app.js file
4. It creates a user registration controller for registering the user.
5. It encrypts a submitted password during the registration.
6. It creates dummy data for being used in the test

#### How should this be manually tested?
1. Start the **Server** and **POSTMAN**.
2. **Browse** the following endpoint route: **localhost:3000/api/auth/signup**.
3. Submit the required information with **POST** method. 
4. Check if the validations work once you don't submit any of the required information.
5. After passing the validations, you should receive a ****201** HTTP Status Code** along with the information that you have submitted.
